src/mark-suspicious-activity-as-reported.ts

```diff
@@ -0,0 +1,109 @@
+} from "@aws-sdk/client-sqs";
+import {
+  DynamoDBDocumentClient,
+  UpdateCommand,
+  QueryCommand,
+} from "@aws-sdk/lib-dynamodb";
+import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import { ActivityLogEntry } from "./common/model";
+
+const dynamoClient = new DynamoDBClient({});
+const dynamoDocClient = DynamoDBDocumentClient.from(dynamoClient);
+
+export const sendSqsMessage = async (
+  messageBody: string,
+  queueUrl: string | undefined
+): Promise<SendMessageCommandOutput> => {
+  const { AWS_REGION } = process.env;
+  const client = new SQSClient({ region: AWS_REGION });
+  const message: SendMessageRequest = {
+    QueueUrl: queueUrl,
+    MessageBody: messageBody,
+  };
+  return client.send(new SendMessageCommand(message));
+};
+
+export const getItemByEventId = async (
+  tableName: string,
+  indexName: string,
+  eventId: string
+): Promise<{ user_id: string; timestamp: number }> => {
+  const getItem = new QueryCommand({
+    TableName: tableName,
+    IndexName: indexName,
+    KeyConditionExpression: "event_id = :event_id",
+    ExpressionAttributeValues: {
+      ":event_id": eventId,
+    },
+  });
+
+  const result = await dynamoDocClient.send(getItem);
+
+  if (result?.Items?.length !== 1) {
+    throw Error(
+      `Expecting exactly 1 result from getItemByEventId, but got ${result?.Items?.length}`
+    );
+  }
+  const item = result.Items[0];
+  return { user_id: item.user_id, timestamp: item.timestamp };
+};
+
+export const markEventAsReported = async (
+  tableName: string,
+  user_id: string,
+  timestamp: number
+) => {
+  const command = new UpdateCommand({
+    TableName: tableName,
+    Key: {
+      user_id,
+      timestamp,
+    },
+    UpdateExpression: "set reported_suspicious = :reported_suspicious",
+    ExpressionAttributeValues: {
+      ":reported_suspicious": true,
+    },
+  });
+
+  return dynamoDocClient.send(command);
+};
+
+export const handler = async (event: SNSEvent): Promise<void> => {
+  const { DLQ_URL, TABLE_NAME, INDEX_NAME } = process.env;
+  await Promise.all(
+    event.Records.map(async (record) => {
+      try {
+        if (!TABLE_NAME) {
+          throw new Error(
+            "Cannot handle event as table name has not been provided in the environment"
+          );
+        }
+        if (!INDEX_NAME) {
+          throw new Error(
+            "Cannot handle event as index name has not been provided in the environment"
+          );
+        }
+        const receivedEvent: ActivityLogEntry = JSON.parse(record.Sns.Message);
+
+        const { user_id, timestamp } = await getItemByEventId(
+          TABLE_NAME,
+          INDEX_NAME,
+          receivedEvent.event_id
+        );
+        await markEventAsReported(TABLE_NAME, user_id, timestamp);
+      } catch (err) {
+        const response = await sendSqsMessage(record.Sns.Message, DLQ_URL);
+        console.error(
+          `[Message sent to DLQ] with message id = ${response.MessageId}`,
+          err as Error
```

Comment:
If we can't send to the DLQ we will also lose the original error that caused us to try to send to the DLQ.  Suggest err is logged first in the catch block.

Reply:
If there is an error that stops us reaching this line of code, then it will throw an error which will be caught in the logs